src/main/java/org/fsg1/fmms/backend/services/CurriculaService.java

```diff
@@ -39,10 +41,14 @@ public String getQueryCurriculaString() {
      *
      * @return A JSON ObjectNode representing an array of curricula.
      */
-    public JsonNode get(final String query, final Object... parameters) throws SQLException, IOException {
+    public JsonNode get(final String query, final Object... parameters) throws SQLException, IOException,
+            EntityNotFoundException {
         final ResultSet resultSet = getConn().executeQuery(query, parameters);
         resultSet.next();
         final String jsonString = resultSet.getString("curricula");
+
+        if (jsonString == null) throw new EntityNotFoundException();
```

Comment:
Why the fuck are you testing jsonString??
resultSet.next() will return false if there is no row to read.
See: https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#next()